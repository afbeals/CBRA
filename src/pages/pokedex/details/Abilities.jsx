// Extenal
import React from 'react';
import PropTypes from 'prop-types';

// Constants
const classname = 'abilities';

// Component
const Abilities = ({ pClassname, moves, abilities }) => (
  <div data-testid="details-abilites" className={`${pClassname}__${classname}`}>
    <div className="section">
      <p className="section__title">abilities</p>
      <ul className="section__list">
        {abilities.map(({ ability: { name } }, index) => (
          <li key={`${name}-${index}`} className="section__list__item">
            {name}
            <span className="passive">passive </span>
          </li>
        ))}
        {moves.map(({ move: { name } }, index) => (
          <li key={`${name}-${index}`} className="section__list__item">
            {name}
          </li>
        ))}
      </ul>
    </div>
  </div>
);

Abilities.propTypes = {
  pClassname: PropTypes.string.isRequired,
  moves: PropTypes.array,
  abilities: PropTypes.array,
};

Abilities.defaultProps = {
  moves: [],
  abilities: [],
};

export default Abilities;

export { Abilities };
