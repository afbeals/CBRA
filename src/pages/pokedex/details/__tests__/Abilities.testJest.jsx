// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import Abilities from '../Abilities';

// tests
describe('Abilities', () => {
  afterEach(cleanup);
  const mockStats = {};
  beforeEach(() => {
    mockStats.abilities = [
      {
        ability: { name: 'chlorophyll' },
      },
    ];
    mockStats.moves = [
      {
        move: {
          name: 'ray',
        },
      },
    ];
  });

  it('should match Abilities to snapshot', () => {
    const { getByTestId } = render(<Abilities pClassname={'parent'} />);
    const abilities = getByTestId('details-abilites');
    expect(abilities).toMatchSnapshot();
  });

  it('should have a passive ability', () => {
    const { getByText } = render(
      <Abilities pClassname={'parent'} abilities={mockStats.abilities} />,
    );
    const passive = getByText('passive');
    const ability = getByText('chlorophyll');
    expect(ability).toBeTruthy();
    expect(passive).toBeTruthy();
  });

  it('should render the passed moves', () => {
    const { getByText } = render(
      <Abilities pClassname={'parent'} moves={mockStats.moves} />,
    );
    const move = getByText('ray');
    expect(move).toBeTruthy();
  });
});
