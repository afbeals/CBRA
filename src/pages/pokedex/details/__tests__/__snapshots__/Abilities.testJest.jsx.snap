// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abilities should match Abilities to snapshot 1`] = `
<div
  class="parent__abilities"
  data-testid="details-abilites"
>
  <div
    class="section"
  >
    <p
      class="section__title"
    >
      abilities
    </p>
    <ul
      class="section__list"
    />
  </div>
</div>
`;
